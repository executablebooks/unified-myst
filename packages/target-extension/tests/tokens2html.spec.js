import { micromark } from 'micromark'

import { mystTargetMmarkExt, mystTargetHtmlExt } from '../src/index.js'

const fixtures = [
    '()=',
    '(abc)',
    '(abc)=',
    ' (abc)=',
    '(abc)= ',
    '(a bc)=',
    String.raw`(ab\\c)=`,
    '(abc)= d',
]

describe('Convert source to HTML', () => {
    test.each(fixtures)(`%j`, (content) => {
        const options = {
            extensions: [mystTargetMmarkExt],
            htmlExtensions: [mystTargetHtmlExt],
        }
        const htmlText = micromark(content, options)
        expect(htmlText).toMatchSnapshot()
    })
})
