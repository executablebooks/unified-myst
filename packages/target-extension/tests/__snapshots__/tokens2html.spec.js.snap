// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert source to HTML " (abc)=" 1`] = `"<a class=\\"anchor\\" id=\\"abc\\" href=\\"#abc\\">ðŸ”—</a>"`;

exports[`Convert source to HTML "()=" 1`] = `"<p>()=</p>"`;

exports[`Convert source to HTML "(a bc)=" 1`] = `"<a class=\\"anchor\\" id=\\"a bc\\" href=\\"#a bc\\">ðŸ”—</a>"`;

exports[`Convert source to HTML "(ab\\\\\\\\c)=" 1`] = `"<a class=\\"anchor\\" id=\\"ab\\\\c\\" href=\\"#ab\\\\c\\">ðŸ”—</a>"`;

exports[`Convert source to HTML "(abc)" 1`] = `"<p>(abc)</p>"`;

exports[`Convert source to HTML "(abc)= " 1`] = `"<a class=\\"anchor\\" id=\\"abc\\" href=\\"#abc\\">ðŸ”—</a>"`;

exports[`Convert source to HTML "(abc)= d" 1`] = `"<p>(abc)= d</p>"`;

exports[`Convert source to HTML "(abc)=" 1`] = `"<a class=\\"anchor\\" id=\\"abc\\" href=\\"#abc\\">ðŸ”—</a>"`;
